adcm_client/objects.py

```diff
@@ -738,11 +738,15 @@ def job_list(self, paging=None, **args) -> "JobList":
 
     @allure_step("Wait for task end")
     def wait(self, timeout=None, log_failed=True):
-        status = self.wait_for_attr("status",
-                                    self._END_STATUSES,
-                                    timeout=timeout)
-        if log_failed and status == "failed":
-            self._log_jobs('failed')
+        try:
+            status = self.wait_for_attr("status",
+                                        self._END_STATUSES,
+                                        timeout=timeout)
+            if log_failed and status == "failed":
+                self._log_jobs('failed')
+        except TimeoutError as e:
+            self._log_jobs()
```

Comment:
If you guard line 746 with if, why you do nothing with line 748? Should log_failed also control timeout?

Reply:
Yes it does. Fixed.